tests/platform/thermal_control_test_helper.py

```diff
@@ -0,0 +1,367 @@
+        fields = get_fields(line, field_ranges)
+        actual_data[fields[0]] = fields
+    
+    return mocker_object.check(actual_data)
+
+
+def check_thermal_algorithm_status(dut, mocker_factory, expected_status):
+    """
+    Check thermal control algorithm status.
+    :param dut: DUT object representing a SONiC switch under test.
+    :param mocker_factory: Mocker factory.
+    :param expected_status: Expect thermal control algorithm status.
+    :return: True if actual thermal control status match expect value.
+    """
+    thermal_mocker = mocker_factory(dut, 'ThermalStatusMocker')
+    if thermal_mocker is not None:
+        return thermal_mocker.check_thermal_algorithm_status(expected_status)
+    return True  # if vendor doesn't provide a thermal mocker, ignore this check by return True.
+
+
+def restart_thermal_control_daemon(dut):
+    """
+    Restart thermal control daemon by killing it and waiting supervisord to restart
+    it automatically.
+    :param dut: DUT object representing a SONiC switch under test.
+    :return:
+    """
+    logging.info('Restarting thermal control daemon...')
+    find_thermalctld_pid_cmd = 'docker exec -i pmon bash -c \'pgrep thermalctld | sort\''
+    output = dut.command(find_thermalctld_pid_cmd)
+    assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
+    assert len(output["stdout_lines"]) == 2, "There should be 2 thermalctld process"
+    pid_0 = int(output["stdout_lines"][0].strip())
+    pid_1 = int(output["stdout_lines"][1].strip())
+    # find and kill the parent process
+    pid_to_kill = pid_0 if pid_0 < pid_1 else pid_1
+    logging.info('Killing old thermal control daemon with pid: {}'.format(pid_to_kill))
+    kill_thermalctld_cmd = 'docker exec -i pmon bash -c \'kill {}\''.format(pid_to_kill)
+    output = dut.command(kill_thermalctld_cmd)  # kill thermalctld and wait supervisord auto reboot thermalctld
+    assert output["rc"] == 0, "Run command '%s' failed" % kill_thermalctld_cmd
+    
+    # make sure thermalctld has restarted
+    max_wait_time = 5
+    while max_wait_time > 0:
+        max_wait_time -= 1
+        output = dut.command(find_thermalctld_pid_cmd)
+        assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
+        if len(output["stdout_lines"]) != 2:
+            time.sleep(1)
+            continue
+        
+        new_pid_0 = int(output["stdout_lines"][0].strip())
+        new_pid_1 = int(output["stdout_lines"][1].strip())
+        parent_pid = new_pid_0 if new_pid_0 < new_pid_1 else new_pid_1
+        
+        if parent_pid == pid_to_kill:
+            logging.info('Old thermal control daemon is still alive, waiting...')
+            time.sleep(1)
+            continue
+        else:
+            logging.info('New pid of thermal control daemon is {}'.format(parent_pid))
+            return
+
+    assert 0, 'Wait thermal control daemon restart failed'
```

Comment:
To avoid impact followed test case, when restarting thermal daemon failed, need to give the upper layer caller an indication so the upper layer can restore the system(eg. config reload).

Reply:
Agree, config reload or reboot as a last chance to restore

Reply:
fixed